<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BCsc vignette • BCscTutorial</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BCsc vignette">
<meta property="og:description" content="BCscTutorial">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BCscTutorial</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BCscvignette.html">BCsc vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BCscvignette_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BCsc vignette</h1>
                        <h4 class="author">Raffaele A Calogero</h4>
            
            <h4 class="date">12/16/2020</h4>
      
      
      <div class="hidden name"><code>BCscvignette.Rmd</code></div>

    </div>

    
    

<div id="section-1-introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#section-1-introduction" class="anchor"></a>Section 1: Introduction</h2>
</div>
<div id="section-2-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#section-2-methods" class="anchor"></a><strong>Section 2</strong>: Methods</h2>
<p>All the methods and analyses described in this thesis were done using R and the Bray Curtis metrics (BCsc) for single cell data analysis was controlled by an rCASC function. All software fro BCsc metrics was implemented in a docker container deposited at repbioinfo repository at docker hub. All calculated data are available as <a href="https://figshare.com/account/home#/projects/94457">figshare project</a>.</p>
<div id="section-2-1-datasets" class="section level3">
<h3 class="hasAnchor">
<a href="#section-2-1-datasets" class="anchor"></a><strong>Section 2.1</strong>: Datasets</h3>
<div id="section-2-1-1-seta-and-setc" class="section level4">
<h4 class="hasAnchor">
<a href="#section-2-1-1-seta-and-setc" class="anchor"></a><strong>Section 2.1.1</strong>: SetA and setC</h4>
<p>SetA and C were prepared to validate rCASC <a href="https://pubmed.ncbi.nlm.nih.gov/31494672/"><em>Alessandri et al. 2019</em></a>. These datasets were generated combining different cell types from Zheng paper <a href="https://pubmed.ncbi.nlm.nih.gov/28091601/"><em>Zheng et al. 2017</em></a>:</p>
<ul>
<li>setA 100 cells randomly selected for each cell type:
<ul>
<li><ol start="2" style="list-style-type: upper-alpha">
<li>B-cells , (M) Monocytes (100K reads/cell), (S) Hematopoietic stem cells , (NK) Natural Killer cells , (N) Naive T-cells</li>
</ol></li>
</ul>
</li>
<li>setC 100 cells randomly selected for each cell type:
<ul>
<li><ol start="3" style="list-style-type: upper-alpha">
<li>Cytotoxic T-cells , (M) Monocytes, (T) T-helper cells, (NK) Natural Killer, (N) Naive T-cells</li>
</ol></li>
</ul>
</li>
</ul>
</div>
<div id="section-2-1-2-contaminated-set-setac30" class="section level4">
<h4 class="hasAnchor">
<a href="#section-2-1-2-contaminated-set-setac30" class="anchor"></a><strong>Section 2.1.2</strong>: Contaminated set: setAc30</h4>
<p>We constructed a contaminated dataset in which a fraction of genes from setA were substituted with a random set of counts, 30%, from setC, i.e.setAc30. In figure , it shown the co-clustering of setA and setAc30.</p>
<pre><code>## Warning: package 'knitr' was built under R version 3.6.2</code></pre>
<div class="figure" style="text-align: center">
<img src="img/fig1.png" alt="Co-clustering of setA and setAc30. A) clusters generated by seurat. B) setA and setAc30 cells" width="80%"><p class="caption">
Co-clustering of setA and setAc30. A) clusters generated by seurat. B) setA and setAc30 cells
</p>
</div>
</div>
<div id="section-2-1-3-set1-and-set2" class="section level4">
<h4 class="hasAnchor">
<a href="#section-2-1-3-set1-and-set2" class="anchor"></a><strong>Section 2.1.3</strong>: Set1 and set2</h4>
<p>Datasets 1 and 2 were derived from two PBMC datasets available at 10XGenomics repository. Set1 was generated using version 3.1 chemistry as instead set2 with version 3.0 chemistry. Both datasets included antibodies labeling. Starting from these datasets we constructed set1 and 2, using the following cells:</p>
<ul>
<li>CD14+CD16- (Monocytes, M), CD56+CD34- (Natural Killer, NK), CD4+CD25+CD127- (T regulatory, TR), CD4+CD45RO+ccr7+ (T memory, TM), CD19+CD20+ (B cell, B), cd8+cd45ra+ccr7+ (naive T, N)</li>
</ul>
<p>100 cells were collected for each cell type in the two experiments. Cells were cleaned up removing all cell with more than 10% mitochondrial genes. Duplicated cells, i.e. those cells showing inconsistent surface makers were removed. Genes were filtered selecting the top 10000 most variant genes and from them the top 5000 most expressed.</p>
<p>set1: 94 M, 95 NK, 85 TR, 62 TM, 96 B, 86 N</p>
<p>set2: 97 M, 94 NK, 84 TR, 82 TM, 92 B, 95 N</p>
</div>
</div>
</div>
<div id="section-3-results" class="section level2">
<h2 class="hasAnchor">
<a href="#section-3-results" class="anchor"></a><strong>Section 3</strong>: Results</h2>
<div id="section-3-1-creation-of-the-contaminated-dataset-setac30" class="section level3">
<h3 class="hasAnchor">
<a href="#section-3-1-creation-of-the-contaminated-dataset-setac30" class="anchor"></a><strong>Section 3.1</strong>: Creation of the contaminated dataset setAc30</h3>
<p>We have transformed setA contaminating it with 30% of the setC counts, randomly selected. In brief, descrivere come hai fatto la contaminazione</p>
</div>
<div id="section-3-2-analysis-of-pbmc_nextgem-and-pbmc_v3_chemistry" class="section level3">
<h3 class="hasAnchor">
<a href="#section-3-2-analysis-of-pbmc_nextgem-and-pbmc_v3_chemistry" class="anchor"></a><strong>Section 3.2:</strong> Analysis of pbmc_NextGEM and pbmc_v3_chemistry</h3>
<p>pbmc_NextGEM dataset is the dataset made of <a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.1.0/5k_pbmc_protein_v3_nextgem">5k Peripheral blood mononuclear cells (PBMCs) from a healthy donor with cell surface proteins (Next GEM)</a> available at 10XGenomics repository.</p>
<p>pbmc_v3_chemistry dataset is the dataset made of <a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.1.0/5k_pbmc_protein_v3">5k Peripheral blood mononuclear cells (PBMCs) from a healthy donor with cell surface proteins (v3 chemistry)</a> available at 10XGenomics repository.</p>
<p>Both datasets were analysed in the following way:</p>
<ul>
<li><p>Low quality cells, i.e. those containing less that 100 genes were removed. All cells with a ribosomal gene percentage between 10 and 70% and a mitochondrial percentage between 1 and 20% were also retained.</p></li>
<li><p>10K most variable genes were selected and out of them 5K most expressed genes were selected.</p></li>
<li><p>Seurat [ref] was used to cluster the data using the first 5 PCA components.</p></li>
<li><p>Cluster specific gene prioritization was done with COMET software [ref].</p></li>
</ul>
<div class="figure" style="text-align: center">
<img src="img/fig2.png" alt="rCASC clustering. A) pbmc NextGEM 10 clusters. B) pbmc NextGEM CSS. C) pbmc v3 chemistry 8 clusters. C) pbmc v3 chemistry CSS" width="80%"><p class="caption">
rCASC clustering. A) pbmc NextGEM 10 clusters. B) pbmc NextGEM CSS. C) pbmc v3 chemistry 8 clusters. C) pbmc v3 chemistry CSS
</p>
</div>
<p>We detected a total of 10 clusters for pbmc_NextGEM, 7 of them with high CSS score, Fig. AB, and 8 clusters for pbmc_v3_chemistry, 6 of them with high CSS, Fig. CD.</p>
<p>These datasets incorporate a panel of 31 TotalSeq™-B antibodies (BioLegend) that were used to assign the cell types for set1 and set2: CD3, CD4, CD8a, CD11b, CD14, CD15, CD16, CD19, CD20, CD25, CD27, CD28, CD34, CD45RA, CD45RO, CD56, CD62L, CD69, CD80, CD86, CD127, CD137, CD197, CD274, CD278, CD335, PD-1, HLA-DR, TIGIT, IgG1, IgG2a, IgG2b.</p>
</div>
<div id="analysis-of-set1-and-set2" class="section level3">
<h3 class="hasAnchor">
<a href="#analysis-of-set1-and-set2" class="anchor"></a>Analysis of set1 and set2</h3>
<div class="figure" style="text-align: center">
<img src="img/fig3.png" alt="Cell type assignment. A) classical monocytes. B) Natural Killer cells. C) pbmc_v3_chemistry 8 clusters. C) T regulatory cells. D) T memory cells. E) B-cells. F) naive T cells" width="80%"><p class="caption">
Cell type assignment. A) classical monocytes. B) Natural Killer cells. C) pbmc_v3_chemistry 8 clusters. C) T regulatory cells. D) T memory cells. E) B-cells. F) naive T cells
</p>
</div>
<p>In Fig. , are summarised the filters applied to the different combinations of Abs to obtain the subsets of cells we were interested in set1 and 2.</p>
<p>After sub-population selection, from each population the first 100 cells were selected or in case of fewer cell, all cell were selected. Only cell characterized by a mitochondrial genes percentage between 1 and 10 were kept.</p>
<div class="figure" style="text-align: center">
<img src="img/fig4.png" alt="rCASC clustering. A) set1 clusters. B) set1 CSS. C) set 2 clusters. D) set2 CSS" width="80%"><p class="caption">
rCASC clustering. A) set1 clusters. B) set1 CSS. C) set 2 clusters. D) set2 CSS
</p>
</div>
<p>Inconsistent cells, i.e. cells associated at more that one cell type were removed. 10K most variable genes were selected and out of them 5K most expressed genes were selected. The remain cells were clustered as described for pbmc_NextGEM and pbmc_v3_chemistry. In both cases we obtained 5 clusters, all with very good stability, Fig. .</p>
<p>The content of all clusters is quite homogeneous but those containing T derived cells. naive T cell clusters are contaminated by Tmem (TM) and Treg (TR) cells. The cluster of TR has some contamination by TM, Fig. .</p>
<div class="figure" style="text-align: center">
<img src="img/fig5.png" alt="Clusters cell type content. A) set1. B) set2." width="80%"><p class="caption">
Clusters cell type content. A) set1. B) set2.
</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by The package maintainer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
